From c6f31ea0b49e012942aa0d90090ba907f4905b4d Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:17 -0400
Subject: [infiniband] ocrdma: do not skip setting deffered_arm

Message-id: <df98c746811ad7180471ecb590e9ad6bebe61f7e.1411571481.git.dledford@redhat.com>
Patchwork-id: 96654
O-Subject: [Patch RHEL7 42/48] RDMA/ocrdma: do not skip setting deffered_arm
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

ib_request_notify_cq() when called for the first time
ocrdma tries to skip setting deffered_arm flag. This
may lead CQ to an un-armed state thus, never generating
CQ event and leaving consumer in hung state.

This patch removes the part of code that skkips setting
deffered_arm flag.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Mitesh Ahuja <mitesh.ahuja@emulex.com>
(cherry picked from commit f93439e476d012b2503dbb07fe0fc675bcbff099)
Signed-off-by: Doug Ledford <dledford@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index 4527311..8f5f257 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -2846,11 +2846,9 @@ int ocrdma_arm_cq(struct ib_cq *ibcq, enum ib_cq_notify_flags cq_flags)
  if (cq->first_arm) {
   ocrdma_ring_cq_db(dev, cq_id, arm_needed, sol_needed, 0);
   cq->first_arm = false;
-  goto skip_defer;
  }
- cq->deferred_arm = true;
 
-skip_defer:
+ cq->deferred_arm = true;
  cq->deferred_sol = sol_needed;
  spin_unlock_irqrestore(&cq->cq_lock, flags);
 
-- 
1.7.1